packages/ember-data/tests/integration/store/query-record-test.js

```diff
@@ -0,0 +1,81 @@
+var Person, store, env;
+var run = Ember.run;
+
+module("integration/store/query-record - Query one record with a query hash", {
+  setup: function() {
+    Person = DS.Model.extend({
+      updatedAt: DS.attr('string'),
+      name: DS.attr('string'),
+      firstName: DS.attr('string'),
+      lastName: DS.attr('string')
+    });
+
+    env = setupStore({
+      person: Person
+    });
+    store = env.store;
+  },
+
+  teardown: function() {
+    run(store, 'destroy');
+  }
+});
+
+test("It raises an assertion when no type is passed", function() {
+  expectAssertion(function() {
+    store.queryRecord();
+  }, "You need to pass a type to the store's queryRecord method");
+});
+
+test("It raises an assertion when no query hash is passed", function() {
+  expectAssertion(function() {
+    store.queryRecord('person');
+  }, "You need to pass a query hash to the store's queryRecord method");
+});
+
+test("When a record is requested, the adapter's queryRecord method should be called.", function() {
+  expect(1);
+
+  env.registry.register('adapter:person', DS.Adapter.extend({
+    queryRecord: function(store, type, query) {
+      equal(type, Person, "the query method is called with the correct type");
+      return Ember.RSVP.resolve({ id: 1, name: "Peter Wagenet" });
+    }
+  }));
+
+  run(function() {
+    store.queryRecord('person', { related: 'posts' });
+  });
+});
+
+test("When a record is requested, and the promise is rejected, .queryRecord() is rejected.", function() {
+  env.registry.register('adapter:person', DS.Adapter.extend({
+    queryRecord: function(store, type, query) {
+      return Ember.RSVP.reject();
+    }
+  }));
+
+  run(function() {
+    store.queryRecord('person', {}).then(null, async(function(reason) {
+      ok(true, "The rejection handler was called");
+    }));
+  });
+});
+
+test("When a record is requested, and the promise is rejected, the record should be unloaded.", function() {
+  expect(2);
+
+  env.registry.register('adapter:person', DS.Adapter.extend({
+    queryRecord: function(store, type, query) {
+      return Ember.RSVP.reject();
+    }
+  }));
+
+  run(function() {
+    store.queryRecord('person', {}).then(null, async(function(reason) {
+      ok(true, "The rejection handler was called");
+    }));
+  });
+
+  ok(!store.hasRecordForId('person', 1), "The record has been unloaded");
```

Comment:
Yep that was an old test from the "fetchQueryOne" method which worked with an id, removing the test